apps/browser/qml/pages/components/TabView.qml

```diff
@@ -40,131 +32,144 @@ SilicaGridView {
     signal closeAllCanceled
     signal closeAllPending
 
-    onCountChanged: if (count > 0) closingAllTabs = false
-    onClosingAllTabsChanged: if (closingAllTabs) closeAllPending()
-
-    width: parent.width - Theme.horizontalPageMargin
-    height: parent.height
-    x: Theme.horizontalPageMargin
-    currentIndex: -1
-    header: PageHeader {
-        //: Tabs
-        //% "Tabs"
-        title: qsTrId("sailfish_browser-he-tabs")
-    }
-    footer: spacer
-    cellHeight: thumbnailHeight + Theme.paddingLarge
-    cellWidth: thumbnailWidth + Theme.paddingLarge
-
-    delegate: TabItem {
-        id: tabItem
-
-        enabled: !closingAllTabs
-        opacity: enabled ? 1.0 : 0.0
-        Behavior on opacity { FadeAnimator {}}
-
-        width: thumbnailWidth
-        height: thumbnailHeight
-
-        GridView.onAdd: AddAnimation {
-            target: tabItem
-        }
-        GridView.onRemove: RemoveAnimation {
-            target: tabItem
-        }
-    }
-
-    // Behind tab delegates
-    children: [
-        PrivateModeTexture {
-            z: -1
-            visible: opacity > 0.0
-            opacity: privateMode ? 1.0 : 0.0
-
-            Behavior on opacity { FadeAnimation {} }
-        },
+    anchors.fill: parent
 
-        MouseArea {
+    header: Private.TabBar {
+        id: header
+        model: headerModel
+        Shared.Background {
+            anchors.fill: parent
             z: -1
-            width: tabView.width
-            height: tabView.height
-            onClicked: hide()
         }
-    ]
-
-    ConfigurationValue {
-        id: showCloseAllAction
-        key: "/apps/sailfish-browser/settings/show_close_all"
-        defaultValue: true
     }
 
-    PullDownMenu {
-        id: pullDownMenu
+    model: [ persistentTabView, privateTabView ]
 
-        flickable: tabView
+    onCurrentIndexChanged: {
+        // TODO: Add remorsePopup
+        // if (remorsePopup) {
+        //     remorsePopup.trigger()
+        // }
+        privateMode = currentIndex !== 0
+    }
+    Component.onCompleted: {
+        persistentIcon.updateGrubImage()
+        privateIcon.updateGrubImage()
+        currentIndex = privateMode
+    }
 
-        MenuItem {
-            text: tabView.privateMode ?
-                    //: Menu item switching back to normal browser
-                    //% "Normal browsing"
-                    qsTrId("sailfish_browser-me-normal_browsing") :
-                    //: Menu item switching to private browser
-                    //% "Private browsing"
-                    qsTrId("sailfish_browser-me-private_browsing")
-            onDelayedClick: {
-                if (remorsePopup) {
-                    remorsePopup.trigger()
-                }
+    PrivateModeTexture {
```

Comment:
Currently the private mode background texture appears/disappears across the whole page as the tab is changed. I'm wondering if maybe it would look better if it slid in along with the page?

Reply:
I agree, corrected